### PR TITLE
fixed code quality issues

### LINKED ISSUE #84: Improve code quality
The current quality rating of the codebase is `B`. Make changes to the repo to make it reach `A`. Please go through [this](https://app.codacy.com/manual/kharsh39/barview-android/issues/index) to know what changes would help.

### PR DESCRIPTION
**Pull request by dino-saurabh**
fixed almost all the code quality issues pertaining to the library.the quality issues which remain are pertaining to documentation of readme ,bugReport and similar files. 

## Issue that this pull request solves

Fixes: #84 

## Proposed changes

Brief description of what is fixed or changed

## Types of changes

_Put an `x` in the boxes that apply_

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [x] Documentation update (Documentation content changed)
-   [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings

### CODE REVIEW

**Comment by dino-saurabh** [comment]
@krharsh some of which require some design decisions to make in which I need your help please review the changes!!

**Comment by dino-saurabh** [comment]
and for the remaining code quality issues which are mainly in documentation part of the project which I think would be better handled by the people who created them since It would be a lot easier for them to fix those issues !!

**Comment by krharsh17** [comment]
There are 3 new issues created by this PR, fix them, please.

File: library/src/main/java/in/krharsh17/barview/BarGroup.java [hunk 1]
```diff
@@ -437,23 +433,26 @@ int dp(float dp) {
      */
     public Drawable getRoundRect(String color,int currentLayer,int totalLayer,int radius) {
 
+        int localRadius=radius;
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarGroup.java]: Why this?

Comment (dino-saurabh) [reply library/src/main/java/in/krharsh17/barview/BarGroup.java]: This was an issue @krharsh17  that we should not assign/ manipulate passed in param values rather we should create local variables for assigning tasks and inner manipulation.

**Comment by dino-saurabh** [comment]
@krharsh17  fixed the 3 new codacy warnings!! review please.

File: library/src/main/java/in/krharsh17/barview/BarGroup.java (hunk 1 above)
Comment (krharsh17) [reply library/src/main/java/in/krharsh17/barview/BarGroup.java]: Oh, okay. And are you sure that this won't break any functionality?

Comment (dino-saurabh) [reply library/src/main/java/in/krharsh17/barview/BarGroup.java]: no I guess it wont you can see the changes below

**Review by krharsh17 — APPROVED** [review]
Looks good to me

**Pull request merged**